File: Go/14/14.go
package longest_common_prefix

import "strings"

func longestCommonPrefix(strs []string) string {
	longestPrefix := strs[0]

	for i := 1; i < len(strs); i++ {
		for strings.Index(strs[i], longestPrefix) != 0 {
			longestPrefix = longestPrefix[:len(longestPrefix)-1]
			if longestPrefix == "" {
				return ""
			}
		}
	}

	return longestPrefix
}

File: C++/src/14.cpp
#include <solution_tmpl.h>
using namespace std;

string Solution::longestCommonPrefix(vector<string> &strs) {
  string longestPrefix = strs[0];

  for (size_t i = 1; i < strs.size(); i++) {
    while (strs[i].find(longestPrefix) != 0) {
      longestPrefix.pop_back();
      if (longestPrefix.empty()) {
        return "";
      }
    }
  }

  return longestPrefix;
}
